### PR TITLE
refactor/use sync-webhook

### PR DESCRIPTION
**Pull request by peelar**
#215

### CODE REVIEW

File: apps/taxes/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "@material-ui/core": "^4.12.4",
     "@material-ui/icons": "^4.11.3",
     "@material-ui/lab": "4.0.0-alpha.61",
-    "@saleor/app-sdk": "0.33.0",
+    "@saleor/app-sdk": "0.0.0-pr-20230307100508",
```
Comment (lkostrowski) [inline apps/taxes/package.json:22]: replace with 0.34.0 before merging

File: apps/taxes/src/modules/taxes/providers/taxjar/taxjar-calculate.ts [hunk 2]
```diff
@@ -5,14 +5,24 @@ import {
...
 
+type FetchTaxesLinePayload = {
+  id: string;
+  quantity: number;
+  taxCode?: string | null;
+  discount: number;
+  chargeTaxes: boolean;
+  unitAmount: number;
+  totalAmount: number;
+};
```
Comment (lkostrowski) [inline apps/taxes/src/modules/taxes/providers/taxjar/taxjar-calculate.ts:24]: where does it come from? Is it your internal typo or something that comes from saleor or taxjar?

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: apps/taxes/src/pages/api/webhooks/order-calculate-taxes.ts [hunk 3]
```diff
@@ -1,27 +1,95 @@
...
+
+  const { data } = validation;
+  logger.info({ data }, "Payload is valid.");
+
+  try {
+    const client = createClient(authData.saleorApiUrl, async () =>
+      Promise.resolve({ token: authData.token })
+    );
+
+    const providersConfig = await new GetProvidersConfigurationService({
```
Comment (krzysztofwolski) [inline apps/taxes/src/pages/api/webhooks/order-calculate-taxes.ts:54]: Would recommend configuration getters as a separate function. Will make handlers' code cleaner and easier to follow

**Review by krzysztofwolski — APPROVED** [review]
LGTM, added code recommendation.

File: .changeset/sour-scissors-wait.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"saleor-app-taxes": minor
+---
+
+replaced copied sync webhook with app-sdk one
```
Comment (lkostrowski) [inline .changeset/sour-scissors-wait.md:5]: This is not a commit message, this is a public changelog. It should be written well. It also should contain entire feature scope, not last change - in this case something that app-sdk was updated, code was removed etc

File: .changeset/sour-scissors-wait.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"saleor-app-taxes": minor
```
Comment (lkostrowski) [inline .changeset/sour-scissors-wait.md:2]: its a patch change, functionality wasnt changed

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: apps/taxes/src/pages/api/webhooks/order-calculate-taxes.ts [hunk 6]
```diff
@@ -1,27 +1,95 @@
...
+  logger.info({ payload }, "Handler called with payload");
 
-  logger.info({ calculatedTaxes }, "Taxes calculated");
-  return res.status(200).json(calculatedTaxes);
-});
+  if (!authData) {
+    logger.error("Auth data not found");
+    logger.info("Returning no data");
+    return res.status(200).json({});
+  }
```
Comment (lkostrowski) [inline apps/taxes/src/pages/api/webhooks/order-calculate-taxes.ts:34]: Webhook utility checks for AuthData, this is redundant

File: apps/taxes/src/pages/api/webhooks/order-calculate-taxes.ts [hunk 7]
```diff
@@ -1,27 +1,95 @@
...
+
+  logger.info("Parsing payload...");
```
Comment (lkostrowski) [inline apps/taxes/src/pages/api/webhooks/order-calculate-taxes.ts:36]: Its rather `debug` not `info`

**Review by lkostrowski — CHANGES_REQUESTED** [review]

**Comment by lkostrowski** [comment]
@peelar lets bump sdk to 0.34.1 (and update changeset to have good version) - it should fix the build
https://github.com/saleor/saleor-app-sdk/pull/207

**Review by lkostrowski — APPROVED** [review]

**Pull request merged**